reference/api/keys.md

```diff
@@ -68,22 +75,59 @@ List all existing API keys. **Expired keys are included in the response**, but d
       "createdAt": "2021-08-11T10:00:00Z",
       "updatedAt": "2021-08-11T10:00:00Z"
     }
-  ]
+  ],
+  "offset":0,
+  "limit":3,
+  "total":7
 }
 ```
 
+::: note
+API keys are displayed in descending order based on their `createdAt` date. This means that the most recently created keys appear first.
+:::
+
 ### Returned fields
 
+Returns API keys in an array called `results`, along with the following fields:
+
+#### `offset`
+
+The number of keys skipped over.
+
+#### `limit`
+
+The maximum number of keys to be returned by the request.
+
+#### `total`
+
+The total number of API keys that can be browsed.
+
+### The `results` array
+
+For each key, it returns:
+
+#### `name`
+
+A human-readable name for the key. Default value is `null`.
+
 #### `description`
 
 A description for the key. Default value is `null`.
 
+#### `uid`
+
+A [uuid v4](https://www.sohamkamani.com/uuid-versions-explained) to identify the API key. If not specified, it is automatically generated by Meilisearch.
+
 #### `key`
 
-An alphanumeric key value generated by Meilisearch on API key creation. Used for authorization when [making calls to a protected Meilisearch instance](/learn/security/master_api_keys.md#communicating-with-a-protected-instance).
+An alphanumeric key value generated by Meilisearch by hashing the `uid` and the master key on API key creation. Used for authorization when [making calls to a protected Meilisearch instance](/learn/security/master_api_keys.md#communicating-with-a-protected-instance).
 
 This value is also used as the `{key}` path variable to [update](#update-a-key), [delete](#delete-a-key), or [get](#get-one-key) a specific key.
 
+::: note
+Since `key` is a hash of the `uid` and master key, `key` values are deterministic between instances sharing the same configuration. This means if the master key changes, all `key` values are automatically changed. The `key` field is computed at runtime and therefore is not propagated to dumps and snapshots. Even if someone does have access to your dumps or snapshot, they cannot make malicious requests.
```

Comment:
What do you think about this?

```suggestion
Since `key` is a hash of the `uid` and master key, `key` values are deterministic between instances sharing the same configuration. This means if the master key changes, all `key` values are automatically changed. 

Since the `key` field is depends on the master key, it is computed at runtime and therefore not propagated to dumps and snapshots. As a result, even if a malicious user comes into possession of your dumps or snapshots, they will not have access to your instance's API keys.
```

Reply:
Thanks Gui!